python/shotgun_desktop/startup.py

```diff
@@ -86,12 +86,26 @@ def __desktop_engine_supports_authentication_module(engine):
 
     :returns: True if the engine supports the authentication module, False otherwise.
     """
-    if engine.version.lower() == 'undefined':
-        logger.warning("The version of the tk-desktop engine is undefined.")
+    if engine.version.lower() == "undefined":
+        logger.warning("The version of the tk-desktop engine is undefined. Assuming authentication is supported.")
         return True
     return LooseVersion(engine.version) >= "v2.0.0"
 
 
+def __desktop_engine_uses_core_logging(engine):
+    """
+    Tests if the engine uses the logging code from core. All versions above 3.0.0 suppor this feature.
+
+    :param engine: The desktop engine to test.
+
+    :returns: True if the engine supports logging using core, False otherwise.
+    """
+    if engine.version.lower() == "undefined":
+        logger.warning("The version of the tk-desktop engine is undefined. Assuming logging is supported.")
+        return True
+    return LooseVersion(engine.version) >= "v3.0.0"
```

Comment:
We'll have bumped the version of the desktop engine since it uses a completely new logging architecture and is incompatible with core 0.17.